open_rarity/resolver/opensea_api_helpers.py

```diff
@@ -257,87 +350,136 @@ def get_collection_with_metadata_from_opensea(
         raise ERCStandardError(
             "We currently do not support non EVM standards at the moment"
         )
+    total_supply = int(stats["total_supply"])
+    tokens = get_all_collection_tokens(
+        slug=opensea_collection_slug,
+        total_supply=total_supply,
+        use_cache=use_cache,
+    )
 
     collection = Collection(
         name=collection_obj["name"],
         attributes_frequency_counts=collection_obj["traits"],
-        tokens=[],
+        tokens=tokens,
     )
 
     collection_with_metadata = CollectionWithMetadata(
         collection=collection,
         contract_addresses=[contract["address"] for contract in contracts],
-        token_total_supply=int(stats["total_supply"]),
+        token_total_supply=max(total_supply, collection.token_total_supply),
```

Comment:
In what cases would `total_supply` not equal `collection.token_total_supply`? Would we want to compute rarity in these cases?

Reply:
Unfortunately, I saw cases where the token total supply was unfortunately wrong :( e.g. in bayc it was 9998 but in actuality its 10k